packages/ui/src/atoms/Link/Link.test.tsx

```diff
@@ -0,0 +1,51 @@
+import { render } from '@testing-library/react'
+import { axe } from 'jest-axe'
+import React from 'react'
+
+import type { LinkProps } from '.'
+import LinkComponent from '.'
+
+const TestLink = ({ href = '/', testId }: LinkProps<'a'>) => {
+  return (
+    <LinkComponent testId={testId} href={href}>
+      Link
+    </LinkComponent>
+  )
+}
+
+describe('Link', () => {
+  describe('Data Attributes', () => {
+    it('`Link` should have `data-store-link` attribute', () => {
+      const { getByTestId } = render(<TestLink />)
+
+      expect(getByTestId('store-link')).toHaveAttribute('data-store-link')
+    })
+
+    it('`Link` should receive a custom component', () => {
```

Comment:
I feel like the main contribution for this test is to verify that the `as` component is being rendered, so I'd leave it outside the **Data attributes** context. I know we're checking for a `data-store-link`, but it's more a safety check.